operations/observability/mixins/workspace/rules/components/nodes/alerts.libsonnet

```diff
@@ -22,6 +22,19 @@
             },
             expr: 'nodepool:node_load1:normalized{nodepool=~".*workspace.*"} > 3',
           },
+          {
+            alert: 'AutoscalerAddsNodesTooFast',
+            labels: {
+              severity: 'critical',
+            },
+            'for': '1m',
+            annotations: {
+              runbook_url: 'https://github.com/gitpod-io/observability/blob/main/runbooks/AutoscalerAddsNodesTooFast.md',
+              summary: "Autoscaler is adding new nodes rapidly",
+              description: 'Autoscaler in cluster {{ $labels.cluster }} is rapidly adding new nodes.',
+            },
+            expr: '((sum(cluster_autoscaler_nodes_count) by (cluster)) - (sum(cluster_autoscaler_nodes_count offset 10m) by (cluster))) > 10',
```

Comment:
[This would have fired a lot recently](https://grafana.gitpod.io/explore?orgId=1&left=%7B%22datasource%22:%22VictoriaMetrics%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22((sum(cluster_autoscaler_nodes_count)%20by%20(cluster))%20-%20(sum(cluster_autoscaler_nodes_count%20offset%2010m)%20by%20(cluster)))%20%3E%2010%22%7D%5D,%22range%22:%7B%22from%22:%22now-60d%22,%22to%22:%22now%22%7D%7D), are all valid? Seems like a lot.

Reply:
Hmmm.
We do seem to scale them up quite aggressively, not sure if this is normal or not:
[grafana link](https://grafana.gitpod.io/explore?orgId=1&left=%7B%22datasource%22:%22VictoriaMetrics%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22(sum(cluster_autoscaler_nodes_count)%20by%20(cluster))%22%7D%5D,%22range%22:%7B%22from%22:%22now-60d%22,%22to%22:%22now%22%7D%7D)

I can tweak alert to be less aggressive, so to catch only really large spikes like the one May 11th. wdyt?

Reply:
Yes, I think that would be a good skateboard. I'm just trying to avoid sending unnecessary pages knowing that we might have to tweak this again in the future.

[15 may be a good number.](https://grafana.gitpod.io/explore?orgId=1&left=%7B%22datasource%22:%22VictoriaMetrics%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22((sum(cluster_autoscaler_nodes_count)%20by%20(cluster))%20-%20(sum(cluster_autoscaler_nodes_count%20offset%2010m)%20by%20(cluster)))%20%3E%2015%22%7D%5D,%22range%22:%7B%22from%22:%22now-60d%22,%22to%22:%22now%22%7D%7D), reduces to 2 incidents.

Reply:
done!